### PR TITLE
Form Consumer now limits hooks and instantiated classes

### LINKED ISSUE #5665: Form Consumer should limit hooks and instantiated classes
Currently, the Service Provider for the `Form\LegacyConsumer` is instantiating classes for callbacks whose hooks never fire.

We are supporting 19 deprecated hooks which are paired with 19 new hooks, each of which are run multiple times in multiple contexts. This creates a problem when each feature spins up a bunch of classes and hooks that aren't necessary.

As an example, the new hooks are used to update the Donation Receipt, which is only need when the donation receipt is being show. Currently, we are setting up each of those hooks when the Donation Receipt isn't even being show. The same goes for the Legacy Template donation confirmation, the email tags, the donation details, the field validation, etc. This adds up very quickly.

## Acceptance Criteria

- [x] `SetupFieldConfirmation` only processes in the context of the Legacy template donation confirmation.
- [x] `SetupFieldEmailTag` only processes in the context for the email editor and sending email.
- [x] `SetupFieldPersistance` only processes in the context of creating a new donation.
- [x] `SetupFieldReciept` only processes in the context of the Sequoia template Receipt.
- [x] `SetupFieldValidation` only processes in the context of rendering fields and validating new donations.
- [x] `SetupPaymentDetailsDisplay` only processes in the context of the donation details screen.

### PR DESCRIPTION
**Pull request by kjohnson**
Resolves #5665 

## Description

Refactors commands classes to only instantiate within the specific hook.

## Testing Instructions

Specific commands should only run within their own hooks. For example, if now showing a receipt, the `SetupFieldReceipt` command should not be instantiated.

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@since` tags included in DocBlocks
-   [x] Changes logged to the `Unreleased` section of `CHANGELOG.md`
-   [x] Includes unit and/or end-to-end tests
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

File: src/Framework/FieldsAPI/FormField/FieldStoreAsMeta.php [hunk 1]
```diff
@@ -17,8 +17,8 @@ trait FieldStoreAsMeta {
 	 * @unreleased
 	 * @return $this
 	 */
-	public function storeAsDonorMeta() {
-		$this->storeAsDonorMeta = true;
+	public function storeAsDonorMeta( $storeAsDonorMeta = true ) {
+		$this->storeAsDonorMeta = $storeAsDonorMeta;
```
Comment (JasonTheAdams) [inline src/Framework/FieldsAPI/FormField/FieldStoreAsMeta.php:21]:
```suggestion
		$this->storeAsDonorMeta = (bool) $storeAsDonorMeta;
```

**Review by JasonTheAdams — APPROVED** [review]
Great job! This has improved performance _and_ pulls the hooks back into the Service Provider. I added one very minor suggestion, but otherwise great! 👍

**Comment by kjohnson** [comment]
@JasonTheAdams the hooks in the command classes always bugged me 😅. It feels good to refactor that code.

File: src/Framework/FieldsAPI/FormField/FieldStoreAsMeta.php (hunk 1 above)
Comment (kjohnson) [reply src/Framework/FieldsAPI/FormField/FieldStoreAsMeta.php:21]: Yeah, that makes sense. I'll get it updated.

Comment (kjohnson) [reply src/Framework/FieldsAPI/FormField/FieldStoreAsMeta.php]: @JasonTheAdams Updated with a test, see https://github.com/impress-org/givewp/pull/5670/commits/9a3753e819c555707f4fe7373862d001d5c28cc5.

**Pull request merged**